src/librustc/ty/instance.rs

```diff
@@ -45,18 +45,18 @@ pub enum InstanceDef<'tcx> {
     CloneShim(DefId, Ty<'tcx>),
 }
 
-impl<'a, 'tcx> TyCtxt<'a, 'tcx, 'tcx> {
+impl<'a, 'tcx> Instance<'tcx> {
     pub fn is_inline_instance(
-        self,
-        instance: &ty::Instance<'tcx>
+        &self,
+        tcx: TyCtxt<'a, 'tcx, 'tcx>
     ) -> bool {
         use hir::map::DefPathData;
-        let def_id = match instance.def {
+        let def_id = match self.def {
```

Comment:
With only `self.def` being accessed, doesn't this method belong on `InstanceDef`? Same for `requires_local_instance`.

Reply:
You are right, I haven't looked to closely.